Sake/Mixer.cs

```diff
@@ -198,27 +198,27 @@ public IEnumerable<IEnumerable<ulong>> CompleteMix(IEnumerable<IEnumerable<ulong
             for (int i = 0; i < inputArray.Length; i++)
             {
                 var currentUser = inputArray[i];
-                var others = new List<ulong>();
+                var others = new List<Input>();
                 for (int j = 0; j < inputArray.Length; j++)
                 {
                     if (i != j)
                     {
                         others.AddRange(inputArray[j]);
                     }
                 }
-                yield return Decompose(currentUser.Select(s => Money.Satoshis(s)), filteredDenominations, maxVsizeCredentialValue).Select(d => (ulong)d.Amount.Satoshi); ;
+                yield return Decompose(currentUser, filteredDenominations, maxVsizeCredentialValue).Select(d => (ulong)d.Amount.Satoshi); ;
             }
         }
 
         /// <param name="myInputsParam">Input effective values. The fee substracted, this is how the code works in the original repo.</param>
         /// <param name="maxVsizeCredentialValue">Maximum usable Vsize that client can get per alice.</param>
-        public IEnumerable<Output> Decompose(IEnumerable<Money> myInputsParam, IEnumerable<Output> denoms, int maxVsizeCredentialValue)
+        public IEnumerable<Output> Decompose(IEnumerable<Input> myInputsParam, IEnumerable<Output> denoms, int maxVsizeCredentialValue)
```

Comment:
The original code in WalletWasabi's master uses no Input class. Unless you're planning to change that, this is not good.